core/consensus/pow/src/lib.rs

```diff
@@ -0,0 +1,438 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Proof of work consensus for Substrate.
+
+use std::sync::Arc;
+use std::thread;
+use client::{
+	BlockOf, blockchain::{HeaderBackend, ProvideCache},
+	block_builder::api::BlockBuilder as BlockBuilderApi, backend::AuxStore,
+};
+use sr_primitives::Justification;
+use sr_primitives::generic::{BlockId, Digest, DigestItem};
+use sr_primitives::traits::{Block as BlockT, Header as HeaderT, ProvideRuntimeApi};
+use srml_timestamp::{TimestampInherentData, InherentError as TIError};
+use pow_primitives::{PowApi, Difficulty, Seal, POW_ENGINE_ID};
+use primitives::H256;
+use inherents::{InherentDataProviders, InherentData};
+use consensus_common::{
+	BlockImportParams, BlockOrigin, ForkChoiceStrategy,
+	well_known_cache_keys::Id as CacheKeyId, Environment, Proposer,
+};
+use consensus_common::import_queue::{BoxBlockImport, BasicQueue, Verifier};
+use codec::{Encode, Decode};
+use log::*;
+
+/// Auxiliary prefix for PoW engine.
+pub const POW_AUX_PREFIX: [u8; 4] = *b"PoW:";
+
+/// Auxiliary data for PoW.
```

Comment:
Again does it refer to auxilary storage? If so, I'd like to directly mention this in the docs since the term "auxilary" is ambiguous and "auxilary storage" is not?

Reply:
Fixed!